doc/usecases/snippet_sharing/UC_edit_snippet.md

```diff
@@ -0,0 +1,19 @@
+# Use Case: Edit configuration snippet
+
+## Summary
+
+Title: Edit configuration snippet
+Scope: Entry Management
+Level: User Goal
+Actors: Authenticated user (snippet owner), Moderator
+Brief: Users can edit their own database entries. Moderators can edit all configuration snippets in the database, also the ones of other users.

```

Comment:
Yes, I had a look at the GitHub API and I don't think there would be a problem in using it, but I'm not sure how much effort it takes to implement the whole registration process in C++ (not really the language that is normally used with oAuth systems). I think there could be some helpful libraries, but that also means more dependencies. Some library for HTTP requests is the minimal requirement though.

For the second point: No idea if GitHub does allow something like this, but that is definitely a total-minor on my list.